ngx_ssdp_dlna.c

```diff
@@ -268,9 +268,9 @@ ngx_ssdp_dlna_read_request(ngx_event_t *rev)
                                                  ngx_ssdp_dlna_module);
 
     do {
-        socklen = sizeof(struct sockaddr_in);
+        socklen = sizeof(struct sockaddr);
```

Comment:
Better to use ```sockaddr_storage``` as container.

Reply:
I'm not really a C developer, just wanted to use the module. Can you submit a PR ?

Reply:
I think your PR is good enough, only this change, probably, can cause problems. I don't remember is  sizeof(struct sockaddr_in) > sizeof(struct sockaddr), but I'm 100% sure that sizeof(struct sockaddr_storage) can keep sockaddr_in and sockaddr_in6 without stack/mem corruption.

Probably you will find this usable for you: https://github.com/rozhuk-im/ssdpd
This ssdpd announcer have IPv4 and IPv6 support , allow to specify net ifaces to announce and many other important settings.
PHP part is very simple to hack.